### PR TITLE
Detect id_rsa permissions problems and bail out early. Fixes gh-1219

### PR DESCRIPTION
**Pull request by rwaldron**
Signed-off-by: Rick Waldron <waldron.rick@gmail.com>

### CODE REVIEW

File: lib/lan-connection.js [hunk 1]
```diff
@@ -48,7 +47,7 @@ LAN.Connection = function(opts) {
   this.ssh = undefined;
 
   if (Tessel.isProvisioned()) {
-    this.auth.privateKey = opts.privateKey || fs.readFileSync(path.join(Tessel.LOCAL_AUTH_KEY));
+    this.auth.privateKey = opts.privateKey || fs.readFileSync(Tessel.LOCAL_AUTH_KEY);
```
Comment (HarleyKwyn) [inline lib/lan-connection.js]: Should we write a common module that reads the auth key and throws the EACCESS error even in this case?. Then we can use it in both `lib/controller.js` and here.

Comment (rwaldron) [inline lib/lan-connection.js]: If the permissions problem exists, no operation will ever reach this code.

File: lib/controller.js [hunk 2]
```diff
@@ -28,6 +29,17 @@ var responses = {
   auth: 'No Tessels Found.'
 };
 
+// Alert the user that they have a permissions problem.
+try {
+  fs.readFileSync(Tessel.LOCAL_AUTH_KEY);
```
Comment (LinusU) [inline lib/controller.js]: Could probably be `fs.accessSync` which wouldn't read the file into memory

Comment (rwaldron) [inline lib/controller.js]: Oh, good call. Let me try that

**Pull request merged**